get_started/simple_pipeline.md

```diff
@@ -0,0 +1,247 @@
+
+
+The code above is one of the simplest examples of Membrane usage. It plays mp3 file through your device's `portaudio`. Let's make it work.
+
+### Prerequisites
+
+First we need to get all libraries that Membrane needs to operate in our case. You can read about them more if you'd like, but for now we'll just jump to installation:
+
+##### Ubuntu
+```bash
+$ apt install clang-format portaudio19-dev ffmpeg libavutil-dev libswresample-dev libmad0-dev
+```
+##### Mac
+```bash
+$ brew install clang-format portaudio ffmpeg libmad pkg-config
+```
+
+### Creating a Project
+
+While installing Elixir you'll get a bunch of useful tools. One of them is [Mix](https://hexdocs.pm/mix/Mix.html). As you can read in its documentation preface:
+> Mix is a build tool that provides tasks for creating, compiling, and testing Elixir projects, managing its dependencies, and more.
+
+Lets use it to create a project for our first Membrane adventure:
+
+```bash
+$ mix new hello --module Hello
+```
+
+Mix generator will create some files for us. Lets take a closer look into two of them:
+
+**mix.exs** - It's an entry file for our mix project, a place where we can configure it, and set-up our dependencies. We'll do it by adding them into `deps` function:
+
+```elixir
+defp deps do
+  [
+	{:membrane_core, "~> 0.7.0"},
+	{:membrane_file_plugin, "~> 0.6.0"},
+	{:membrane_portaudio_plugin, "~> 0.7.0"},
+	{:membrane_ffmpeg_swresample_plugin, "~> 0.7.1"},
+	{:membrane_mp3_mad_plugin, "~> 0.7.0"}
+  ]
+end
+```
+Next we can fetch them by running
+```bash
+$ mix deps.get
+```
+
+**lib/hello.ex** - Here we'll be creating our pipeline.
+
+### Our first Pipeline
+
+#### Pipeline behaviour
+
+Let's start with declaring that we'll be using `Membrane.Pipeline` behaviour:
+
+```elixir
+defmodule Hello do
+
+  use Membrane.Pipeline
+  
+end
+```
+
+Using a behaviour means we are treating our module as a Membrane pipeline, so we've access to functions defined in `Membrane.Pipeline` module, and we can implement some of it's callbacks.
+Let's implement a first one, `handle_init/1`. As you can see in [the documentation](https://hexdocs.pm/membrane_core/Membrane.Pipeline.html#c:handle_init/1) `handle_init` takes one argument wich can be any type - it's a way to pass some arguments needed for pipeline to start. As our app one and only purpose id to play an mp3 file we can assume that the only value we need to pass into pipeline is a path for a file we want to play:
+```elixir
+defmodule Hello do
+
+  use Membrane.Pipeline
+
+  @impl true
+  def handle_init(path_to_mp3) do
+  end
+  
+end
+```
+
+The main purpose of `handle_init` callback is to prepare our pipeline. Preparing among others means that we need to specify all its elements as children and set-up linkings between them. All other options that can be specified if needed you can find in `Membrane.ParentSpec` [docs](https://hexdocs.pm/membrane_core/Membrane.ParentSpec.html).
```

Comment:
I think Membrane.ParentSpec is not always used while returning from the Pipeline callback - [callback_return_t()](https://hexdocs.pm/membrane_core/Membrane.Pipeline.html#t:callback_return_t/0) describes the return type, which consists of [Membrane.Pipeline.Action.t()](https://hexdocs.pm/membrane_core/Membrane.Pipeline.Action.html#t:t/0) which can be forward_t(), spec_t(), log_metadata_t() and some more - and Membrane.ParentSpec.t() is used probably only in spec_t().

It might be worth describing that there are other actions that can be taken while returning from the callback (and provide the second link I have put there within that description)

Reply:
I suggest writing something like this:

The main purpose of the `handle_init` callback is to prepare our pipeline. Preparing means that we need to specify all its elements as children and set up links between those children to describe the order in which data will flow through the pipeline. 
Pipeline's callbacks are expected to return a status and an optional list of actions to be taken. The action can be of one of the following [types]( https://hexdocs.pm/membrane_core/Membrane.Pipeline.Action.html#t:t/0).
Since we want to spawn children processes and link them, we will use the [`spec_t()` action](https://hexdocs.pm/membrane_core/Membrane.Pipeline.Action.html#t:spec_t/0) which is described with the use of `Membrane.ParentSpec` structure.